docs/userguide/mpi_apps.rst

```diff
@@ -40,116 +73,275 @@ to the desired number of blocks.
 Configuring the Executor
 ++++++++++++++++++++++++
 
-Configure the executor to launch a number of workers equal to the number of MPI tasks per block.
-First set the ``max_workers`` to the number of MPI Apps per block.
-then set ``cores_per_worker=1e-6`` to prevent HTEx from reducing the number of workers
-if you request more workers than cores.
-
-If you plan to only launch one App per block, you are done!
+Here are the steps for configuring the executor:
 
-If not, the executor may need to first partition nodes into distinct groups for each MPI task.
-Partitioning is necessary if your MPI launcher does not automatically ensure MPI task gets exclusive nodes.
-Resources vary in how the list of available nodes is provided,
-but they typically are found as a "hostfile" referenced in an environment variable (e.g., ``PBS_NODEFILE``).
-We recommend splitting this hostfile for the host block into hostfiles for each worker
-by adding code like the following to your ``worker_init``:
+1. Set ``HighThroughputExecutor(enable_mpi_mode=True)``
+2. Set ``HighThroughputExecutor(mpi_launcher="LAUNCHER")`` to one from ("srun", "aprun", "mpiexec")
+3. Set the ``max_workers`` to the number of MPI Apps you expect to run per scheduler job (block).
+4. Set ``cores_per_worker=1e-6`` to prevent HTEx from reducing the number of workers if you request more workers than cores.
```

Comment:
Why wouldn't setting MPI mode do this automatically?

Reply:
The intent here is to make the user aware of the fact that by default we will generally not provision more workers than cores on a compute node. I am wary of trying to set an alternate set of preset values for mpi_mode that takes away configurability when we do not know how users will use this.

Reply:
That's a good idea. Overloading a node drastically could lead to performance degradation and annoyed users too.

Reply:
(feel free to resolve this)